### DISCUSSION TITLE
wlan1 VPN tunnel option automatically turn Off

### DISCUSSION THREAD

**Discussion by Dbhatt20** [Mygod/VPNHotspot#397] [Q&A]
Kudos to you for this project. I have only one issue I am using this on my MIUI Android 12 mobile and sometime when I enabled the VPN tun0 routing via VPN hotspot the option disable automatically. I have set the Upstream and Fallback to VPN tun0 only but still when option got automatically disable the network leak to my Public IP instead of VPN. 

I don't know if possible or not if we have a feature to stop routing when tun0 is not a upstreaming the traffic because this will save us to leak out of public IP to connected device.

**Comment by Mygod** [comment]
I don't understand what you mean. There is not a tun0 routing setting in my app.

**Reply by ReactiveThings** [reply]
@Mygod I think @Dbhatt20 means to have a kill switch functionality. Currently when you disable and enable Hotspot in Android settings manually we share wifi connection without using VPN. In my case wlan1 option is disabled in VPN Hotspot app. 

We need to enable it again manually to use VPN connection. 

Additionally sometimes that option is disabled automatically ( I do not how and why. I have not figured it out yet. ) so my Public IP leaks incidentally.

Can we add functionality to not share wifi connection without VPN? Or Enable (in my case wlan1) option by default when we enable Hotspot in Android Settings?

**Reply by Mygod** [reply]
This is supported via the monitor feature. You could find it in top right of the app when hotspot is enabled.

**Reply by Dbhatt20** [reply]
Awesome, I was not aware about this monitor feature Thanks @Mygod you rocks.....
